_posts/2021-02-09-an-introduction-to-soliditys-fuzz-testing-approach.md

```diff
@@ -0,0 +1,164 @@
+---
+layout: post
+published: true
+title: 'An Introduction to Solidity's Fuzz Testing Approach'
+date: '2021-02-09'
+author: Bhargava Shastry
+category: Explainers
+---
+
+Security vulnerabilities and bugs detract from software quality.
+To discover them early, we have adopted fuzz testing: feeding randomly generated programs to the Solidity compiler and observing the compilation runtime and code generated.
+
+Since Q1 2019, the Solidity compiler is fuzz tested via Google's [open-source software fuzz][1] (oss-fuzz) framework.
+
+In this post, we briefly describe the work that has been done on this front, and work that is currently in progress.
+
+### Fuzzer Overview
```

Comment:
Not sure where to put that best, but maybe mention that the compiler makes very heavy use of assertions which are then hit by the fuzzer?

Reply:
Actually you already mention that "led to an assertion failure" - maybe just extend on that?
Perhaps also mention that bugs in the frontend almost never lead to invalid code because of that.

Reply:
Done.